front/app/components/UI/Breadcrumbs/index.tsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import { colors } from 'utils/styleUtils';
+
+// components
+import { Box, Text } from '@citizenlab/cl2-component-library';
+import Link from 'utils/cl-router/Link';
+
+// types
+type SingleBreadcrumbProps = {
+  label: string;
+  linkTo?: string;
+  isCurrentPage?: boolean;
+};
+
+type BreadcrumbsProps = {
+  breadcrumbs: SingleBreadcrumbProps[];
+};
+
+const SingleBreadcrumb = ({
+  label,
+  linkTo,
+  isCurrentPage,
+}: SingleBreadcrumbProps) => {
+  const SingleBreadcrumbWrapper = ({ children }) => {
+    return (
+      <Box display="inline">
+        <Text color="adminSecondaryTextColor" as="span" fontSize="m">
+          {children}
+        </Text>
+      </Box>
+    );
+  };
+
+  const Divider = () => {
+    return (
+      <Box display="inline" mr="16px" ml="16px">
+        <Text color="adminSecondaryTextColor" as="span" fontSize="m">
+          /
+        </Text>
+      </Box>
+    );
+  };
```

Comment:
Would keep components in their own file. Also, a SingleBreadcrumb can just be Breadcrumb, right? :-)

Reply:
I find the current code a bit high maintenance for what it does, and could be easier to wrap your head around. I would keep a BreadCrumb in 1 file, then import this into the BreadCrumbs file. I would also make the parent container (Breadcrumbs) display: flex, then you won't need display: inline on the children.

And with styles on the parent of a breadcrumb, you shouldn't have to redefine the styles on each component, right?

Reply:
I would keep a BreadCrumb in 1 file, then import this into the BreadCrumbs file. I would also make the parent container (Breadcrumbs) display: flex, then you won't need display: inline on the children.

And with styles on the parent of a breadcrumb, you shouldn't have to redefine the styles on each component, right?

Reply:
The whole file is < 100 lines so I don't think it needs to be split, and moving the single breadcrumb to another file doesn't make sense to me as they are never going to be used separately, a single breadcrumb is just a subcomponent of the parent component and won't ever be used in isolation. 

I named it `SingleBreadcrumb` to make it easy to tell apart from `Breadcrumbs`, instead of just one `s` difference between the two.

I can switch it to flex if you'd like but I think functionally it's the same, it's not really a layout component that needs to handle other states, I felt like putting one breadcrumb inline next to the others was fine

Reply:
I'll let [the code speak](https://github.com/CitizenLabDotCo/citizenlab/compare/CL-956-create-breadcrumbs-component...CL-956/review?expand=1). :-) What do you think about the difference?

I find it easier to understand and there's less repetition (maintenance).
* I do get the argument that this is a mini component so it would be overkill to move the file out here (unless we want to make it more reusable in the future and let the dev pass a styled Breadcrumb as a prop). But then I would just render the component directly in the map and not put it above in a separate component. Much easier to reason about if you don't need to move up the file again, no? https://alexkondov.com/tao-of-react/#avoid-nested-render-functions
* isCurrentPage is a bit of a confusing name to me. isLastBreadcrumb communicates clearer what we want to know (and it shouldn't be part of the props passed to the breadcrumb if I understand its intention well).
* There's also the SingleBreadcrumbWrapper, which could be directly in the code. The conditional part can go inside the component, it's not needed to repeat SingleBreadcrumbWrapper, Divider, etc. Although I think we should just always require linkTo, this will be easier when creating the data we pas as breadcrumbs props.

Speaking of which, have we thought about how we'll get the data for the breadcrumbs prop? That's also part of this task imo, because it'll influence how we design the component (props).

Reply:
I don't really think the old code was difficult to reason about or difficult to maintain, at 92 lines of a purely presentational component. Other than a couple of comments below I think your code is also fine, if you'd prefer to check it in instead please do so.

I don't necessarily agree with that article's point in this context. The argument for having `Divider` defined within `SingleBreadcrumb` is that it doesn't take in props or need to access any variables, it's simply a part of `SingleBreadcrumb` that is needed in two spots in the code, so it's defined in one place and then used as a named component in the two places it's needed.

`isLastBreadcrumb` is a better name for that variable but I believe anyone reading the previous code would have been able to figure out what `isCurrentPage` meant quickly: `isCurrentPage={index === breadcrumbs.length - 1}` is a pretty common convention for checking if something is the final item in an array, and the last breadcrumb generally does represent the "current page" the user is on.

`LinkTo` needs to be optional as the final breadcrumb in a set of breadcrumbs is plain text and not a link, as the link would be to the same page the user is on and therefore kind of useless. In your code the dev is required to supply a link for the final breadcrumb, which is subsequently disabled. I don't think it makes sense to require devs to supply a link that can't be clicked, then disable it/disallow users from clicking on it, rather than just having the final breadcrumb, a non-link with the title of the current page, displayed as text with no link associated.

As for the data, I think it makes sense to wait for someone to use this component in the context of building out the associated feature. Once a dev needs this component they'll be able to see pretty quickly if the current state of entering breadcrumb data matches their needs, and then they can either update it to work for the container's actual needs or use it as-is. 

In general when I code I'm also trying to balance our velocity and our ability to get features out quickly against all of the other tradeoffs. I think my original code was fine for this purpose, as it fulfilled the requirements, was relatively simple and straightforward, and could have been immediately implemented by the dev building the container component for the rest of the feature, which I think is the ultimate goal here. Since it did function as intended, I think trying to fully optimize this component and its code for the purposes of theoretical future maintenance, at this point in the development cycle, is a kind of premature optimization that doesn't necessarily benefit us or our ability to deliver working features.

Reply:
> Other than a couple of comments below I think your code is also fine, if you'd prefer to check it in instead please do so.

Could you copy paste (just 1 file)? Will be easier than merge conflicts.  

> LinkTo needs to be optional as the final breadcrumb in a set of breadcrumbs is plain text and not a link, as the link would be to the same page the user is on and therefore kind of useless. In your code the dev is required to supply a link for the final breadcrumb, which is subsequently disabled. I don't think it makes sense to require devs to supply a link that can't be clicked, then disable it/disallow users from clicking on it, rather than just having the final breadcrumb, a non-link with the title of the current page, displayed as text with no link associated.

I reckon we're going to get them programatically, not manually on each page? So that's why I thought it'll be easier to provide the link to the current page and disable (also with the added benefit it simplifies the component logic). But what you say makes sense, it's weird to pass a link then disable. It'll depend on how we'll fetch the links to previous pages (keep that as simple as possible), so that's why I wanted to make it part of this ticket. It influences how we design the component.  

> As for the data, I think it makes sense to wait for someone to use this component in the context of building out the associated feature. Once a dev needs this component they'll be able to see pretty quickly if the current state of entering breadcrumb data matches their needs, and then they can either update it to work for the container's actual needs or use it as-is.

Usage will be the same everywhere in the admin I believe.